ClinGen/clincoded
# Connects to #943. VCI summary page.

**Known issues:**
1. While on the Summary page, clicking on the browser's **Back** button instead of the "Return to Interpretation" button will cause the page to lose its interpretation state (e.g. such as taking the user back to the "Evidence Only" view).

**Technical details:**
1. The provisional form on the summary page can be saved even when the modified pathogenicity and its reason fields have `null` values.
2. In the use case above, we are at least storing a string value in the `provisional-variant` object for the calculated pathogenicity and a boolean value in the `interpretation` object for the provisional interpretation checkbox.
3. The displayed **Calculated Pathogenicity** uses the in-app state as its primary data source and the stored value as its secondary data source. The in-app state allows the most current calculated value to be displayed, while the stored value provides a fallback in case the in-app state is not present.
4. The displayed **Modified Pathogenicity** uses only the stored `alteredClassification` value in the `provisional-variant` object, regardless the changed selection on the **Modified Pathogenicity** dropdown.
5. The displayed **Provisional Interpretation Status** string is solely determined by the stored boolean value of the provisional interpretation checkbox in the `interpretation` object.
6. When making a selection from the **Modify Pathogenicity** dropdown, a reason needs to be provided in the textarea below. And vice versa. Otherwise, the form can not be submitted.
7. If no associated disease, selecting **Likely Pathogenic** or **Pathogenic** from the **Modify Pathogenicity** will automatically uncheck the provisional checkbox (if it was previously checked) while disabling the checkbox.
8.  If no associated disease, a calculated pathogenicity of **Likely Pathogenic** or **Pathogenic** will automatically uncheck the provisional checkbox (if it was previously checked) while disabling the checkbox.
9. In the **Criteria "Not yet evaluated"** table, the list includes `not-evaluated` criteria stored into the `evaluation` object as well as criteria that have not been evaluated and stored in the db.

**Steps to test:**
1. Login and use 15333 variant_id at the variation selection interface.
2. Start a new interpretation without associating with a disease.
3. Evaluate some criteria so that the calculated pathogenicity is either **Likely Pathogenic** or **Pathogenic**. Click on the **View Summary** button and expect to see the **"Mark as Provisional Interpretation (optional):"** checkbox being disabled.
4. Then select either **Benign**, **Likely Benign** or **Uncertain Significance** from the **Modify Pathogenicity** and expect to see the checkbox being enabled. Also expect to see the **"Explain reason(s) for change"** textarea being highlighted in pink now in alerting the user to provide a reason while the **Save** button is disabled.
5. Now enter some text in the textarea and expect to see the pink highlight goes away as well as the **Save** being enabled.
6. Next, check the provisional interpretation checkbox and save the form. Expect to see a status message displayed as a feedback for the form submission.
7. Further, change the **Modify Pathogenicity** dropdown selection to either **No Selection**, **Likely Pathogenic** or **Pathogenic**. Expect to see the provisional interpretation checkbox being unchecked and disabled.
8. Finally, click the **Return to Interpretation** button and associate the interpretation with a disease. Then click on the **View Summary** button again to return to the summary view. Expect to see the provisional interpretation checkbox being enabled regardless of the calculated/modified pathogenicity.
9. While on the summary page, expect to see the 3 tables listing all the criteria that meet an evaluation strength, were evaluated as "Not met", as well as were "Not yet evaluated".
